ocaml/typing/jkind.ml

```diff
@@ -1009,18 +1164,11 @@ end
 (******************************)
 (* relations *)
 
-let equate_or_equal ~allow_mutation (l1 : t) (l2 : t) =
-  match l1.jkind, l2.jkind with
-  | Any, Any -> true
-  | Immediate64, Immediate64 -> true
-  | Immediate, Immediate -> true
-  | Sort s1, Sort s2 -> (
-    match Sort.equate_tracking_mutation s1 s2 with
-    | (Equal_mutated_first | Equal_mutated_second) when not allow_mutation ->
-      Misc.fatal_errorf "Jkind.equal: Performed unexpected mutation"
-    | Unequal -> false
-    | Equal_no_mutation | Equal_mutated_first | Equal_mutated_second -> true)
-  | (Any | Immediate64 | Immediate | Sort _), _ -> false
+let equate_or_equal ~allow_mutation
+    { jkind = { layout = lay1; externality = ext1 }; history = _ }
+    { jkind = { layout = lay2; externality = ext2 }; history = _ } =
+  Layout.equate_or_equal ~allow_mutation lay1 lay2
+  && Externality.equal ext1 ext2
```

Comment:
Should there be `Jkind_.equate_or_equal` that this just calls?

Reply:
Yes. Good idea.